tsl/src/nodes/decompress_chunk/decompress_chunk.c

```diff
@@ -883,143 +946,155 @@ ts_decompress_chunk_generate_paths(PlannerInfo *root, RelOptInfo *chunk_rel, con
-		 * that would have invalid references due to our
-		 * EquivalenceClasses.
+		 * Add the paths to the chunk relation.
 		 */
-		if (IsA(compressed_path, BitmapHeapPath) && compressed_path->param_info)
-			continue;
+		ListCell *decompressed_cell;
+		foreach (decompressed_cell, decompressed_paths)
+		{
+			Path *path = lfirst(decompressed_cell);
+			add_path(chunk_rel, path);
+		}
+	}
 
+	/* create parallel paths */
+	List *uncompressed_paths_with_parallel =
+		list_concat(uncompressed_table_parallel_pathlist, uncompressed_table_pathlist);
+	foreach (compressed_cell, compressed_rel->partial_pathlist)
+	{
+		Path *compressed_path = lfirst(compressed_cell);
+		List *decompressed_paths = build_on_single_compressed_path(root,
+																   chunk,
+																   chunk_rel,
+																   compressed_path,
+																   add_uncompressed_part,
+																   uncompressed_paths_with_parallel,
+																   &sort_info,
+																   compression_info);
 		/*
-		 * Filter out all paths that try to JOIN the compressed chunk on the
-		 * hypertable or the uncompressed chunk
-		 * Ideally, we wouldn't create these paths in the first place.
-		 * However, create_join_clause code is called by PG while generating paths for the
-		 * compressed_rel via generate_implied_equalities_for_column.
-		 * create_join_clause ends up creating rinfo's between compressed_rel and ht because
-		 * PG does not know that compressed_rel is related to ht in anyway.
-		 * The parent-child relationship between chunk_rel and ht is known
-		 * to PG and so it does not try to create meaningless rinfos for that case.
+		 * Add the paths to the chunk relation.
 		 */
-		if (compressed_path->param_info != NULL)
+		ListCell *decompressed_cell;
+		foreach (decompressed_cell, decompressed_paths)
 		{
-			if (bms_is_member(chunk_rel->relid, compressed_path->param_info->ppi_req_outer))
-				continue;
-			/* check if this is path made with references between
-			 * compressed_rel + hypertable or a nesting subquery.
-			 * The latter can happen in the case of UNION queries. see github 2917. This
-			 * happens since PG is not aware that the nesting
-			 * subquery that references the hypertable is a parent of compressed_rel as well.
-			 */
-			if (bms_overlap(parent_relids, compressed_path->param_info->ppi_req_outer))
-				continue;
+			Path *path = lfirst(decompressed_cell);
+			add_partial_path(chunk_rel, path);
+		}
+	}
 
-			ListCell *lc_ri;
-			bool references_compressed = false;
-			/*
-			 * Check if this path is parameterized on a compressed
-			 * column. Ideally those paths wouldn't be generated
-			 * in the first place but since we create compressed
-			 * EquivalenceMembers for all EquivalenceClasses these
-			 * Paths can happen and will fail at execution since
-			 * the left and right side of the expression are not
-			 * compatible. Therefore we skip any Path that is
-			 * parameterized on a compressed column here.
-			 */
```

Comment:
I think I fixed this some time ago, we shouldn't be creating EquivalenceMembers on compressed columns anymore because they don't make sense anyway. Removed this check and the tests for older issues still pass.